### PR TITLE
fix(core): combine serial and parallel execution + forward sigint to child process

### PR DESCRIPTION
**Pull request by caioaao**
Attempt at fixing #13766 

Combined parallel and sync process spawning since parallel one was already dealing with capturing signals, so it seemed like the easiest way to fix it

### CODE REVIEW

**Comment by luxaritas** [comment]
Is this also related to (or even fixes) #11782?

**Comment by caioaao** [comment]
@luxaritas this fixes the `run-commands` executor, but I didn't check how the `run-script` executor is related

File: packages/nx/src/executors/run-commands/run-commands.impl.ts [hunk 1]
```diff
@@ -208,6 +216,7 @@ function createProcess(
     const processExitListener = () => childProcess.kill();
     process.on('exit', processExitListener);
     process.on('SIGTERM', processExitListener);
+    process.on('SIGINT', () => childProcess.kill('SIGINT'));
```
Comment (AgentEnder) [inline packages/nx/src/executors/run-commands/run-commands.impl.ts:219]: Lets reuse the `childProcessExitListener` here

**Review by AgentEnder — APPROVED** [review]
This looks like a good change, but there are some weird formatting issues. I'll reformat it and address the minor comment I had, but this should be merged later today :)

**Comment by AgentEnder** [comment]
Huh, on second glance I don't have permission to push up to the branch on your fork. Can you run `nx format` after addressing the minor comment?

**Comment by luxaritas** [comment]
Any chance we can get run-script addressed now as well?

**Comment by caioaao** [comment]
@AgentEnder oh that's weird. I thought contributors had push access to forks. Will do!

**Comment by caioaao** [comment]
@luxaritas I'm currently traveling, so I don't think I'll have time for going through `run-scripts` too. I might have time after Jan 23rd though

File: packages/nx/src/executors/run-commands/run-commands.impl.ts (hunk 1 above)
Comment (caioaao) [reply packages/nx/src/executors/run-commands/run-commands.impl.ts:219]: the problem is the exit listener won't forward the signal used to kill the main process. I can parameterize the `processExitListener` but I don't know if it's worth it?

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**